sunpy/net/attr.py

```diff
@@ -60,6 +113,63 @@ def collides(self, other):
     def __eq__(self, other):
         return dict(vars(self)) == dict(vars(other))
 
+    @classmethod
+    def update_values(cls, adict):
+        """
+        This is the method that clients will use to register their values for subclasses of `~sunpy.net.attr.Attr`.
+
+        The input has be a dictionary, with each key being a subclass of Attr.
+        The value for each key should be a list of tuples with each tuple of the form (`Name`, `Description`).
+        If you do not want to add a description, you can put `None` or an empty string.
+        We sanitize the name you provide by removing all special characters and making it all lower case.
+        If it still invalid we will append to the start of the name to make it a valid attribute name.
+        This is an ugly workaround, so please try to pick a valid attribute name.
+
+        **It is recommended that clients register values for all attrs used by their `_can_handle_query` method.**
+
+        Paramaters
+        ----------
+
+        adcit : `dict`
+            A dictionary that has keys of `~sunpy.net.attr.Attr`.
+            Each key should have a list of tuples as it value.
+            Each tuple should be a pair of strings.
+            First string is the attribute name you want to register
+            Second string is the description of the attribute.
+
+        Example
+        -------
+        >>> from sunpy.net import attr, attrs # doctest: +SKIP
+        >>> attr.Attr.update_values({attrs.Instrument: [('AIA', 'AIA is in Space.'), ('HMI', 'HMI is next to AIA.')]}) # doctest: +SKIP
+        >>> attr.Attr._attr_registry[attrs.Instrument] # doctest: +SKIP
+        attr(name=['aia', 'hmi'], name_long=['AIA', 'HMI'], desc=['AIA is in Space.', 'HMI is next to AIA.']) # doctest: +SKIP
+        >>> attr.Attr._attr_registry[attrs.Instrument].name # doctest: +SKIP
+        ['aia', 'hmi']
+        >>> attr.Attr._attr_registry[attrs.Instrument].name_long # doctest: +SKIP
+        ['AIA', 'HMI']
+        >>> attr.Attr._attr_registry[attrs.Instrument].desc # doctest: +SKIP
+        ['AIA is in Space.', 'HMI is next to AIA.']
+        >>> attrs.Instrument.aia, attrs.Instrument.hmi # doctest: +SKIP
+        (<Instrument('AIA')>, <Instrument('HMI')>)
+        """
+        for k, v in adict.items():
+            # Does this make sense?
+            if isiterable(v) and not isinstance(v, str):
```

Comment:
I *think* `isiterable` specifically excludes strings. so.

```suggestion
            if isiterable(v):
```